src/resource.jl

```diff
@@ -332,8 +332,30 @@ function serve_file(
     content_encoding::String;
     buffer::Vector{UInt8} = Vector{UInt8}(undef, 2*1024*1024),
 )
-    size = filesize(path)
-    HTTP.setheader(http, "Content-Length" => string(size))
+    content_length = filesize(path)
+    startbyte, stopbyte = 0, content_length-1
+
+    # Support single byte ranges
+    range_string = HTTP.header(http, "Range")
+    if !isempty(range_string)
+        m = match(r"^bytes=(\d*)-(\d*)$", range_string)
+        if m === nothing
+            @error "malformed HTTP Range header, ignoring" range_string
```

Comment:
```suggestion
            @error "unsupported HTTP Range request, ignoring" range_string
```
The regex doesn't work for multipart range requests, which are valid but not supported, so this covers that too.